tests/nightly/model_backwards_compatibility_check/train_mxnet_legacy_models.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#Author: Piyush Ghai
+
+run_models() {
+	echo '=========================='
+	echo "Running training files and preparing models"
+	echo '=========================='
+	python model_backwards_compat_train.py
+	echo '=========================='
+}
+
+install_mxnet() {
+	version=$1
+	echo "Installing MXNet "$version
+	pip install mxnet==$version
+}
+
+install_boto3(){
+	echo "Intalling boto3"
+	pip install boto3
+}
+
+echo `pwd`
+cd tests/nightly/model_backwards_compatibility_check
+echo `pwd`
+
+install_boto3
+
+## Fetch the latest release tags, filtering out 'rcs' and filtering out some other irrelevant ones
+## This list is sorted in descending order chronologically. Keeping n = 5 for a precautionary check.
+## Sample output for the below git tag command is : 1.2.0 utils 1.1.0 1.0.0 0.12.1
+previous_versions=($(git tag --sort=-creatordate | grep --invert-match rc | head -n 5))
+count=0
+for version in ${previous_versions[*]}
+do
+	# We just need to train the previous two versions. This logic can be changed later on as welll.
+	if [[ "$count" -gt 1 ]]
```

Comment:
Could you elaborate? We have to train everything that's missing.

Reply:
I see, thank you. Wasn't it part of the initial design that we test everything under the same major version, independent of the number of releases?

Reply:
Good point and it's been taken care of in the inference scripts. They pull off all the model files (they don't filter out on which version files were trained on). So in a way, after a couple of releases the training script would have built a pool of all the previous releases for the inference files to work with.

Reply:
This line -- "previous_versions=($(git tag --sort=-creatordate | grep --invert-match rc | head -n 5))" is version agnostic, which means even if you change the major version, the models will still be trained. 
With multiple minor versions, the training will happen on all of the tagged git versions. 

So essentially, whenever a new release (be it major version/minor version) comes out, it will be pulled in using the git-tag command and the training file will save a bunch of models for it in S3. 

Please let me know if there's still some confusion here.

Reply:
I have updated the readme and added more comments to this file in : 1f63941e9ade0eaa2f4361ed5877bc97defcf6fb

Reply:
As stated in-person, we have to train everything under the same major version.

Reply:
Done. d8fa75d72bd87466112f58865cebb38883b63171.